iceoryx_posh/test/integrationtests/service_discovery/test_roudi_findservice.cpp

```diff
@@ -40,6 +40,70 @@ TEST_F(RoudiFindService_test, OfferSingleMethodServiceSingleInstance)
     ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
 }
 
+TEST_F(RoudiFindService_test, OfferServiceWithDefaultDescription)
+{
+    senderRuntime->offerService(iox::capro::ServiceDescription());
+    this->InterOpWait();
+    iox::runtime::InstanceContainer instanceContainer;
+    receiverRuntime->findService(iox::capro::ServiceDescription(), instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    senderRuntime->stopOfferService(iox::capro::ServiceDescription());
+}
+
+TEST_F(RoudiFindService_test, ReOfferSingleMethodServiceSingleInstance)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    iox::runtime::InstanceContainer instanceContainer;
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
+
+    senderRuntime->stopOfferService({"service1", "instance1"});
+    this->InterOpWait();
+    instanceContainer.clear();
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(0u));
+
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
+}
+
+TEST_F(RoudiFindService_test, OfferSameServiceMultipleTimes)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->offerService({"service1", "instance1"});
+
+    this->InterOpWait();
+    InstanceContainer instanceContainer;
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
+}
+
+TEST_F(RoudiFindService_test, FindServiceMultipleTimes)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+
+    this->InterOpWait();
+    InstanceContainer instanceContainer;
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
```

Comment:
What is the goal of this test? That `findService` works even after it was called?

Reply:
yeah

Reply:
I'm not sure if this is something that makes sense. You have the same instance twice in the instanceContainer. This is because we just push but don't clear or check the container if this instance is already in? 
Having the same instance twice in the instance container is very dangerous IMHO. We should fix this!